### ISSUE TITLE
Change foreign key signature in queries

### ISSUE THREAD

**Issue by phreed** [babeloff/fql#9]
Change from -> to = in foreign_key clauses in queries

The foreign_key signature in AqlQuery 
```
queryForeignSig
  : schemaForeignId RARROW LBRACE queryPathMapping+ RBRACE ;
```

**Comment by phreed** [comment]
It appears that the right side of the RARROW (being replaced with EQUAL) is of some kind maybe a transform?
```
queryForeignSig
  : schemaForeignId EQUAL LBRACE transformLiteralSection RBRACE ;
```

**Comment by wisnesky** [comment]
Yes.  In fact, a query Q : S -> T is the same thing as a functor T -> S^{op}-Inst, i.e., each target table T gets an instance Q(T), each target foreign key fk : a -> b gets a transform Q(fk) : b -> a, and each target attribute att : a -> b gets a term Q(att) : b in instance Q(a).

But, inside of queries we want to use "=" instead of "->" to make the transform appear to be a "join condition".  

More generally, the AQL grammar reuses instances and transforms a lot.
